From 8841981021b5f98522ad44fe40e11d0c0e756534 Mon Sep 17 00:00:00 2001
From: Brian Mayer <bleemayer@gmail.com>
Date: Thu, 2 Jul 2026 14:51:51 -0300
Subject: [PATCH] arm64: dts: rockchip: add HYM8563 RTC to Radxa CM5 IO board

The Radxa CM5 IO carrier has a Haoyu HYM8563 real-time clock on I2C6 at
address 0x51, with its interrupt line on GPIO0_B0 and a CR1220 backup
battery holder. Without a device tree node, no /dev/rtc* appears even
when CONFIG_RTC_DRV_HYM8563 is enabled.

Describe the RTC, wire the interrupt pin with a pull-up, mark it as a
wakeup source, and point the rtc0 alias at it so RTC_HCTOSYS/SYSTOHC use
the external clock. Wiring matches Radxa's downstream board support.

Signed-off-by: Brian Mayer <bleemayer@gmail.com>
Tested-by: Brian Mayer <bleemayer@gmail.com>
---
 .../dts/rockchip/rk3588s-radxa-cm5-io.dts     | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588s-radxa-cm5-io.dts b/arch/arm64/boot/dts/rockchip/rk3588s-radxa-cm5-io.dts
index af4a9bc01..1a2c72831 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s-radxa-cm5-io.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-radxa-cm5-io.dts
@@ -19,6 +19,7 @@ / {
 	aliases {
 		ethernet0 = &gmac1;
 		mmc1 = &sdmmc;
+		rtc0 = &hym8563;
 	};
 
 	chosen {
@@ -198,6 +199,18 @@ usbc0_dp_altmode_mux: endpoint {
 			};
 		};
 	};
+
+	hym8563: rtc@51 {
+		compatible = "haoyu,hym8563";
+		reg = <0x51>;
+		#clock-cells = <0>;
+		clock-output-names = "hym8563";
+		interrupt-parent = <&gpio0>;
+		interrupts = <RK_PB0 IRQ_TYPE_LEVEL_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&rtc_int>;
+		wakeup-source;
+	};
 };
 
 &i2s5_8ch {
@@ -221,6 +234,12 @@ usbc0_int: usbc0-int {
 		};
 	};
 
+	hym8563 {
+		rtc_int: rtc-int {
+			rockchip,pins = <0 RK_PB0 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+
 	hdmi {
 		hdmi0_tx_on_h: hdmi0-tx-on-h {
 			rockchip,pins = <4 RK_PB6 RK_FUNC_GPIO &pcfg_pull_none>;

base-commit: 87320be9f0d24fce67631b7eef919f0b79c3e45c
-- 
2.54.0

